### ISSUE TITLE
Fan Control loses control of one fan every few hours until I change a setting

### ISSUE THREAD

**Issue by Rings4** [Rem0o/FanControl.Releases#2590]
Version: 136
Windows 10 21H2
(I know this is an old version, but I haven't had any issue this whole time until only recently and would love to know what's wrong)

Ever since a few weeks ago, Fan Control has been "letting go" or losing control of one of my case fans (3-pin), which I used to set to 0RPM since it started making rattling noise. This was causing it to default back to the fan curve set in the BIOS, making it rattle.

After trying everything I could, I finally changed the curve in the BIOS to keep the fan at 0.

Yesterday I got a new fan (4-pin) to replace my rattling fan, and thought that would be the end of it. The BIOS power mode for the fan is set to Auto, which chooses PWM mode for this fan. The old fan was set to Voltage.

Today I noticed my new fan is spinning faster than it should be, not respecting my fan curve, meaning it defaulted back to the BIOS fan curve. I increased the offset by 1% and it went back to respecting my Fan Control fan curve.

My other 2 case fans work no problem. I can switch them off in Fan Control and see their speeds adjust to the BIOS fan curve, meaning Fan Control had been controlling them the whole time just fine. It's just the one case fan that's losing control, and I don't know why.

I did not update Fan Control for a long time. I'm on V136. My philosophy is if it ain't broke, don't fix it. I also haven't installed any Windows 10 updates since Jan 29, 2023. Updates are completely disabled, and I just checked my update history to confirm. So I'm not sure why I'm suddenly having this issue. Nothing changed. I thought my case fan was dying, which was strange because it hadn't been spinning for a while so its condition shouldn't have gotten worse, but since I replaced it I thought that would be the end of it. Yet here I am, with the same issue.

I set the "Force Apply" setting for now but that setting didn't help with the old fan, and I doubt it'd help here. It's just a matter of time before it randomly loses control again. Any help is appreciated!

**Comment by Rings4** [comment]
> "Force apply" does basically what changing the offset does, so it should help in your situation.
> 
> Just so you understand how FanControl works, the program basically sets a value in memory to tell the fan what speed to rotate. It sets it once then does nothing else. It doesn't "actively" control a fan, it's a set and forget kind of thing. If the fan deviate from that last command FanControl sent, it means "something else" wrote a command at that address. Most of the time it is BIOS fan configuration. FanControl cannot prevent other programs to set a value.
> 
> Is that one fan configured exactly the same in the BIOS as the other 2?

Thank you for the prompt reply. Yes, that fan is set up exactly the same as the other fans. In my BIOS (Gigabyte), you can set one fan's curve, then "Apply To" other fans, which I did. I wish I could tell what was causing fan control to lose control of this one fan in particular.

**Comment by Rings4** [comment]
Unfortunately, setting Force Apply doesn't work. The fan has already ramped back up to the BIOS fan curve.

**Comment by Rem0o** [comment]
Do you get a "?" queston mark next to the % value of your control card?

**Comment by Rings4** [comment]
> Do you get a "?" queston mark next to the % value of your control card?

No, I did at one point with the old fan, once, but never with this new one.

![image](https://github.com/Rem0o/FanControl.Releases/assets/32050292/799064a7-5cf6-4bf7-9092-4e4454960606)

**Comment by Rem0o** [comment]
With you being on V136, it's hard for me to help you out.  Mind sharing a screenshot when the problem occurs?

**Comment by Rings4** [comment]
> With you being on V136, it's hard for me to help you out. Mind sharing a screenshot when the problem occurs?

Sure, I'll post a screenshot next time it happens. I'm getting the feeling it happens whenever the curve is linear since it's not sending a new value to the fan, which is still strange considering the other 2 fans are also linear below a certain temp. I'll post back once it happens again.

**Comment by Rings4** [comment]
> With you being on V136, it's hard for me to help you out. Mind sharing a screenshot when the problem occurs?

It's finally doing it again. Here's a screenshot. I won't change any settings in case you need me to check something else while the fan is spinning at an incorrect speed.

![image](https://github.com/Rem0o/FanControl.Releases/assets/32050292/0501919e-07a0-417e-bd58-33a67c8893e4)

**Comment by ionGL** [comment]
Here is an idea. Backup your current v136 config file and give the latest version a try. If you're not happy, get back to v136 and that backed up config.

**Comment by Rings4** [comment]
> Here is an idea. Backup your current v136 config file and give the latest version a try. If you're not happy, get back to v136 and that backed up config.

Definitely worth a shot. I'll give that a try. Thanks!

**Comment by Rings4** [comment]
> Here is an idea. Backup your current v136 config file and give the latest version a try. If you're not happy, get back to v136 and that backed up config.

Okay, installed the latest version V194. Still happening.

![Screenshot 2024-06-29 01 27 13 FanControl](https://github.com/Rem0o/FanControl.Releases/assets/32050292/79ca8d79-526f-4faf-b247-1c6b6afbc768)

**Comment by Rings4** [comment]
![Screenshot 2024-06-29 14 45 11 FanControl](https://github.com/Rem0o/FanControl.Releases/assets/32050292/42a39193-3405-40ff-b680-d7f31c378367)
Happened again an hour ago.

![Screenshot 2024-06-29 15 44 14 FanControl](https://github.com/Rem0o/FanControl.Releases/assets/32050292/4af8e97c-aee4-40cb-9b48-8e10ccbae0e5)
And once more just now.

**Comment by Rings4** [comment]
I'm unfortunately still having this issue, but I think changing my fan curve to where the flat parts are no longer flat (ie, 40℃ to 63℃ used to be 53% flat, now I made it 52% at 40℃ and 53% at 63℃) has alleviated this issue so far.

Still would be nice to be able to use static values for certain parts of my fan curves again like I've been doing for the past few years.

**Comment by Rem0o** [comment]
From the BIOS screenshot you sent, isn't that a custom fan curve set in the BIOS? This curve will most likely still hit when the temperature touches the big spike up at the far right of it.   Any way to completely disable it or set a fixed fan speed in there?

**Comment by Rings4** [comment]
> From the BIOS screenshot you sent, isn't that a custom fan curve set in the BIOS? This curve will most likely still hit when the temperature touches the big spike up at the far right of it. Any way to completely disable it or set a fixed fan speed in there?

I can set a fixed fan speed in the BIOS, but the strange thing is I've had these custom fan curves for years, long before I started using Fan Control, and they have never interfered with the program before. Even still, only the top exhaust fan is having issues. The other 2 fan headers are working just fine.

The strangest thing is that when the fan starts speeding up, it doesn't even respect the fan curve that is set in the BIOS. It spins much faster than it would spin if the BIOS were controlling it. So I'm not entirely sure the BIOS fan curve is the issue here.

**Comment by ionGL** [comment]
New idea. Set this new problematic fan to voltage, like the old one.

**Comment by ionGL** [comment]
New idea. Set this new problematic fan to voltage, like the old one.
Also i would try this after control mode:voltage. Temp interval: max possible. Then set all curve points to 0% in bios. Let fan control cook.

**Comment by Rings4** [comment]
> New idea. Set this new problematic fan to voltage, like the old one. Also i would try this after control mode:voltage. Temp interval: max possible. Then set all curve points to 0% in bios. Let fan control cook.

Thanks for the new idea. I set the fan to voltage but sadly it's still randomly speeding up for seemingly no reason. The temp interval is already on max possible. I'll set the curve points to 0% in the BIOS when I get home and see what happens. Will write back.

**Comment by Rem0o** [comment]
@Rings4  So was it the fan curve in the BIOS doing the spike behavior you described?

**Comment by Rings4** [comment]
> @Rings4 So was it the fan curve in the BIOS doing the spike behavior you described?

Since I changed the fan curves in my BIOS to 0%, the issue hasn't happened as often, but it still has happens infrequently. It would set the speed to about 67% when it should be 39%. I kind of fixed the issue by setting a very slight curve that goes from 40.6℃ at 39% to 62℃ at 40%, since I think the issue happens when the fans aren't being told to change speed.

I don't know why this one specific fan does that. It's as if something else is controlling it, but I wasn't able to find anything else that could possible be the culprit.

**Comment by adamgs69** [comment]
Hi guys, I'm having a similar problem as well. Every now and then FanControl just seems to lose control of a fan - it has a question mark beside the fan that has no control and it always seems to be a "calibrated" fan. 

Anyways, just throwing my hat in the ring to say that this still seems to be an issue. Version V228 using net 8.
